src/solvers/dgsem_tree/indicators_1d.jl

```diff
@@ -24,6 +24,115 @@ function create_cache(typ::Type{IndicatorHennemannGassner}, mesh,
     create_cache(typ, equations, dg.basis)
 end
 
+# Modified indicator for ShallowWaterEquations1D to apply full FV method on cells
+# containing some "dry" LGL nodes. That is, if an element is partially "wet" then it becomes a
+# full FV element.
+#
+# TODO: TrixiShallowWater: move new indicator type
+function (indicator_hg::IndicatorHennemannGassnerShallowWater)(u::AbstractArray{<:Any, 3
+                                                                                },
+                                                               mesh,
+                                                               equations::ShallowWaterEquations1D,
+                                                               dg::DGSEM, cache;
+                                                               kwargs...)
+    @unpack alpha_max, alpha_min, alpha_smooth, variable = indicator_hg
+    @unpack alpha, alpha_tmp, indicator_threaded, modal_threaded = indicator_hg.cache
+    # TODO: Taal refactor, when to `resize!` stuff changed possibly by AMR?
+    #       Shall we implement `resize!(semi::AbstractSemidiscretization, new_size)`
+    #       or just `resize!` whenever we call the relevant methods as we do now?
+    resize!(alpha, nelements(dg, cache))
+    if alpha_smooth
+        resize!(alpha_tmp, nelements(dg, cache))
+    end
+
+    # magic parameters
+    threshold = 0.5 * 10^(-1.8 * (nnodes(dg))^0.25)
+    parameter_s = log((1 - 0.0001) / 0.0001)
+
+    # If the water height `h` at one LGL node is lower than `threshold_wet`
+    # the indicator sets the element-wise blending factor alpha[element] = 1
+    # via the local variable `indicator_wet`. In turn, this ensures that a pure
+    # FV method is used in partially wet cells and guarantees the well-balanced property.
+    #
+    # Hard-coded cut-off value of `threshold_wet = 1e-4` was determined through many numerical experiments.
+    # Overall idea is to increase robustness when computing the velocity on (nearly) dry cells which
+    # could be "dangerous" due to division of conservative variables, e.g., v = hv / h.
+    # Here, the impact of the threshold on the number of cells being updated with FV is not that
+    # significant. However, its impact on the robustness is very significant.
+    # The value can be seen as a trade-off between accuracy and stability.
+    # Well-balancedness of the scheme on partially wet cells with hydrostatic reconstruction
+    # can only be proven for the FV method (see Chen and Noelle).
+    # Therefore we set alpha to one regardless of its given maximum value.
+    threshold_wet = 1e-4
```

Comment:
I find it slightly confusing that this `threshold_wet` is not the same as `equations.threshold_wet`.

Reply:
That is true. This is cutoff to determine if an element is partially wet and should not be changed. We could rename it here and in `indicators_2d.jl` to be something along the lines of `threshold_partially_wet = 1e-4`.

Reply:
That sounds like a good idea :+1: